syntax = "proto3";

package nighthawk.client;

import "google/protobuf/duration.proto";

// TODO(oschaaf): Ultimately this will be a load test specification. The fact that it
// can arrive via CLI is just a concrete detail. Change this to reflect that.
message CommandLineOptions {
  // See :option:`--rps` for details.
  uint64 requests_per_second = 1;
  // See :option:`--connections` for details.
  uint64 connections = 2;
  // See :option:`--duration` for details.
  google.protobuf.Duration duration = 3;
  // See :option:`--timeout` for details.
  google.protobuf.Duration timeout = 4;
  // See :option:`--h2` for details.
  bool h2 = 5;
  // See :option:`--concurrency` for details.
  string concurrency = 6;
  // See :option:`--verbosity` for details.
  string verbosity = 7;
  // See :option:`--output-format` for details.
  string output_format = 8;
  // See :option:`--prefetch-connections` for details.
  bool prefetch_connections = 9;
  // See :option:`--uri` for details.
  string uri = 10;
}
